### PR TITLE
add pure julia wrapper

### PR DESCRIPTION
**Pull request by aTrotier**

### CODE REVIEW

**Comment by aTrotier** [comment]
Stuff to do :
- remove dimensions with 1 in excess

File: src/BartIO.jl [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
+Define the path to the BART toolbox (store in ENV["TOOLBOX_PATH"])
+"""
+function set_bart_path(pathToBart::String)
+    ENV["TOOLBOX_PATH"]=pathToBart
+    return pathToBart
+end
+
+function get_bart_path()
+        # Check bart toolbox path
+        bart_path = ENV["TOOLBOX_PATH"];
```
Comment (JakobAsslaender) [inline src/BartIO.jl:22]: You don't need semi-colons at the end of the line and, I think, it is discouraged in Julia.

File: src/BartIO.jl [hunk 2]
```diff
@@ -23,27 +49,59 @@ export wrapper_bart
...
-    PyCall.py"""
-    import os
-    import sys
-    os.environ['TOOLBOX_PATH'] = $pathtobart
-    path = os.environ["TOOLBOX_PATH"] + "/python/"
-    sys.path.append(path)
-    """
+    run(`$shell_cmd $cmd_split $infiles $outfiles`)
+
+    output = Array{ComplexF32}[]
```
Comment (JakobAsslaender) [inline src/BartIO.jl:80]: You already know the size of `output`, right? In this case, I think it is better to define this as something like `Vector{Array{ComplexF32}}(undef, length(output))` and then  index in `output[idx]` in the for loop.

File: src/BartIO.jl [hunk 3]
```diff
@@ -23,27 +49,59 @@ export wrapper_bart
...
+    for idx in 1:nargout
```
Comment (JakobAsslaender) [inline src/BartIO.jl:81]: convert to `idx in eachindex(output, outfiles)` (together with above change) -- this is considered better practice for safety/speed.

File: src/BartIO.jl [hunk 4]
```diff
@@ -78,8 +141,8 @@ function read_cfl(filename::String)
     for i in eachindex(data)
         data[i] = read(fid, Float32) + 1im * read(fid, Float32)
     end
-
     close(fid)
+    data = reshape(data,dims...)
```
Comment (JakobAsslaender) [inline src/BartIO.jl:145]: Does this line do anything? `data = Array{ComplexF32}(undef, Tuple(dims))` already creates `data` in the right format, doesn't it?

**Review by JakobAsslaender — COMMENTED** [review]
Wow, that was quick! This looks great, thanks so much! I made a few (mostly stylistic) suggestions.

**Comment by aTrotier** [comment]
Thanks for the comment. I thought the work will be more difficult than using PyCall. It is way better this way :)

When the first release will be up. I have to change the Benchmark scripts

**Pull request merged**